uint8 command
uint8 pulse_mode